src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -989,9 +1196,16 @@ private static BoundNode Rewrite(ImmutableDictionary<BoundExpression, (Nullabili
                                             analyzedNullabilityMapOpt,
                                             snapshotBuilderOpt);
 
+            finalNullableState = null;
             try
             {
                 Analyze(walker, symbol, diagnostics, initialState, snapshotBuilderOpt);
+                // we only go to the trouble of saving the final state if we're analyzing field initializers
+                if (symbol is MethodSymbol method && method.IsConstructor() && initialState is null && !useConstructorExitWarnings)
```

Comment:
>symbol is MethodSymbol method && method.IsConstructor() && initialState is null && !useConstructorExitWarnings [](start = 20, length = 110)

Should the caller pass in a `bool generateFinalState`?

Reply:
Done